SilKit/cmake/get_compiler_arch.cmake

```diff
@@ -80,73 +80,93 @@ function(get_linux_distro outDistroName outDistroVersion)
 endfunction()
 
 function(get_compiler_arch outCompiler outArch outPlatform  )
-    #get arch
+    # compute bitness
+
     if("${CMAKE_SIZEOF_VOID_P}" STREQUAL "8")
         set(SYSTEM_BITNESS "64")
     elseif("${CMAKE_SIZEOF_VOID_P}" STREQUAL "4")
         set(SYSTEM_BITNESS "32")
     else()
-        message(FATAL_ERROR "Bitness is not supported: \"${CMAKE_SIZEOF_VOID_P}\"")
+        message(FATAL_ERROR "Unsupported pointer size: ${CMAKE_SIZEOF_VOID_P}")
     endif()
-    message(STATUS "SIL Kit using ${SYSTEM_BITNESS}-Bit build")
 
-    #get OS
+    message(STATUS "SIL Kit using ${SYSTEM_BITNESS}-bit build")
+    message(STATUS "SIL Kit using compiler version ${CMAKE_CXX_COMPILER_ID} ${CMAKE_CXX_COMPILER_VERSION}")
+
+    # compute platform and architecture
+
+    set(_platform "${CMAKE_SYSTEM_NAME}")
+    set(_system_arch "UNKNOWN")
+
     if("${CMAKE_SYSTEM_NAME}" STREQUAL "Windows")
         set(_platform "Win")
 
-        set(_system_arch "x86")
+        # building on (or for) Windows-on-ARM will lead to invalid architecture values
+
         if("${SYSTEM_BITNESS}" STREQUAL "64")
             set(_system_arch "x86_64")
+        else()
+            set(_system_arch "x86")
         endif()
+    elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "QNX")
+        set(_platform "QNX")
+        set(_system_arch "${SILKIT_TARGET_ARCHITECTURE}")
     elseif(UNIX)
         # generic unix has uname
         get_uname(_un_name _un_machine)
         set(_platform "${_un_name}")
         set(_system_arch "${_un_machine}")
-    else()
-        set(_system_arch "${CMAKE_SYSTEM_NAME}")
     endif()
 
     if("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux")
+        # on linux the platform tag is used for the distribution name and version
         get_linux_distro(_distro_name _distro_version)
         if( DEFINED _distro_name AND DEFINED _distro_version)
             set(_platform "${_distro_name}-${_distro_version}")
         endif()
     endif()
 
-    # get toolset/compiler id + version
+    # compute compiler name
+
     set(_tool_tag "UNKNOWN")
-    if(UNIX)
-        set(_id "${CMAKE_CXX_COMPILER_ID}")
+
+    if(MINGW)
+        # currently any mingw compiler uses the tag MinGW for the compiler
+        set(_tool_tag "MinGW")
+    else()
         if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
-            set(_id "gcc")
+            set(_tool_tag "gcc")
         elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
-            set(_id "clang")
-        endif()
-        message(STATUS "Using compiler version: ${_id}-${CMAKE_CXX_COMPILER_VERSION}")
-        set(_tool_tag ${_id})
-    elseif(MINGW)
-        set(_tool_tag "MinGW")
-    elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "Windows")
-        silkit_validate_preset_toolset()
-        if(CMAKE_VS_PLATFORM_TOOLSET MATCHES "v140")
```

Comment:
I think CMAKE_VS_PLATFORM_TOOLSET was used for very old cmake versions, but I think we're safe to deprecate that.
The `MSVC_TOOLSET_VERSION` is safe to use since cmake Version 3.12

Reply:
The top-level `CMakeLists.txt` has had the minimum version set to 3.12 for quite a while already.